extensions/wikia/ImageReview/ImageReviewSpecialController.class.php

```diff
@@ -236,39 +324,67 @@ private function getOrderingMethod() {
 	}
 
 	private function parseAction() {
-		$actions = explode( '/', $this->getPar() );
-		return array_pop( $actions );
+		$pathParts = explode( '/', $this->getPar() );
+		
+		// If there are path parts, try to find the action.  Explode *always* returns an
+		// array even if the content its "exploding" is empty or null.  In those cases it
+		// bizarrely returns an array with one null item.  So just check that the first
+		// element of this array is not empty.
+		if ( !empty( $pathParts[0] ) ) {
+			$action = array_pop( $pathParts );
+			return empty( self::ALLOWED_ACTIONS ) ? self::ACTION_UNREVIEWED : $action;
+		}
+
+		// Return the default action
+		return self::ACTION_UNREVIEWED;
 	}
 
 	private function checkUserPermissions() {
-		$this->accessQuestionable = $this->wg->User->isAllowed( 'questionableimagereview' );
-		$this->accessRejected = $this->wg->User->isAllowed( 'rejectedimagereview' );
-
 		// check permissions
-		if (!$this->specialPage->userCanExecute($this->wg->User)) {
+		if ( !$this->specialPage->userCanExecute( $this->wg->User ) ) {
 			$this->specialPage->displayRestrictionError();
 			return false;
 		} elseif ( $this->action == self::ACTION_QUESTIONABLE && !$this->accessQuestionable ) {
-			$this->specialPage->displayRestrictionError( 'questionableimagereview' );
+			$this->specialPage->displayRestrictionError();
 			return false;
 		} elseif ( $this->action == self::ACTION_REJECTED && !$this->accessRejected ) {
-			$this->specialPage->displayRestrictionError( 'rejectedimagereview' );
+			$this->specialPage->displayRestrictionError();
+			return false;
+		} elseif ( in_array( $this->action, [ self::ACTION_STATS, self::ACTION_CSVSTATS ] ) &&
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This branch's code block is the same as the block for the branch on line 344. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1871)